### ISSUE TITLE
support pacman in library loading and installation

### ISSUE THREAD

**Issue by xhdong-umd** [rstudio/rsconnect#156]
I like to use [`pacman` ](https://github.com/trinker/pacman)for loading packages, because I can write this and share my code to others. 

```
if (!require("pacman")) install.packages("pacman")
pacman::p_load(shiny, DT, data.table)
```

They can just run the code and install packages they don't have automatically. If I used `library()` they will need to find every missing package and install them manually before the script can work.

However this doesn't work when deployed to shinyapps.io. Because shinyapps only look for `library` to install packages, and there is a explicit `install.packages` call in the script which is not allowed in shinyapps.

Right now I'm deploying my shiny apps both in shinyapps.io and github. People can run my app in github directly with `shiny::runGitHub`. Because of this limit, I cannot use the pacman lines in my code which make the github deployment less perfect.

Is it possible to extend shinyapps to support this usage? It should not be hard, just

1. ignore the install.packages line since it will never success anyway
2. scan the packages in this script. Just add `pacman` and all packages appeared in `p_load` call to the packages dependency list.

Thanks!

**Comment by grayskripko** [comment]
any progress, guys?

**Comment by jmcphers** [comment]
We don't plan to do this; package installation should not be a side effect of running a Shiny app.

**Comment by xhdong-umd** [comment]
In real life we often need to deal with this kind of nuisances especially with less tech savvy users. Side effect sound bad but we are dealing with side effect all the time, without side effect nothing will happen inside a user interface -- all the changes in data are pure and silent and invisible.

I can understand the decision, though the cumbersome plumbing job still need to be done, just in a less easy way.

@grayskripko you actually don't need to wait for this be implemented. The package scanning job is done by packrat, so I implemented a patch to support pacman usage in rstudio/packrat#360 above. Just install [my fork](https://github.com/xhdong-umd/packrat), and your app can be deployed and detected properly. There are some limitations, see trinker/pacman#82 for details.

I implemented this in 2017, now my app no longer need this as I built my app into a package. However pacman is still convenient for fast prototyping shiny apps and deploy to users.
